### PR TITLE
Add Dockerfile and build image flow

### PR DESCRIPTION
**Pull request by grieve54706**
## Description
Add Dockefile and GitHub workflow to build images.

## Additional information
This PR uses `pull_request` `labeled` event to test. Please ignore the many `test` label messages.
```yaml
on:
  pull_request:
    types: [labeled]
```
We may use the event `on.push.branches` to test the GitHub workflow in the future.

### CODE REVIEW

File: ibis-server/README.md [hunk 1]
```diff
@@ -1,16 +1,59 @@
...
+Install the dependencies
+```bash
+make install
+```
+Run the server
+```bash
+make run
+```
+
+## Contributing
```
Comment (goldmedal) [inline ibis-server/README.md:40]:
```suggestion
## Developer Guide
```
It's a kind of guide to show how start a development environment for developers.
I think `Developer Guide` would be better.

File: ibis-server/Makefile [hunk 2]
```diff
@@ -13,3 +13,9 @@ dev:
 
 test:
 	poetry run pytest
+
+docker-build:
+	docker image build . -t wren-engine-ibis
+
+docker-run:
```
Comment (goldmedal) [inline ibis-server/Makefile:20]:
```suggestion
dev-docker-run:
```
How about add a `dev` prefix? I guess someone maybe confuse on which version would be started by this command. The image on repo? or The local build.
I guess this one will start the local one, right?

**Review by goldmedal — APPROVED** [review]
Some minor comments, Others LGTM.

**Review by paopa — APPROVED** [review]
LGTM 👍

**Comment by goldmedal** [comment]
Thanks @grieve54706  @paopa

**Pull request merged**